### PR TITLE
UCR sliding datespan

### PR DESCRIPTION
**Pull request by nickpell**
implements http://manage.dimagi.com/default.asp?162937

Uses the filter type "sliding_date" to represent a date that corresponds to the sliding date range.

Same UI for saving date ranges as in conventional reports.

@czue @NoahCarnahan 

To-do:
- [ ] Use properly formatted datepicker widget in saved reports modal
- [x] Prevent user from adding more than one filter of type "sliding_date"
- [ ] Add tests.

### CODE REVIEW

File: corehq/apps/reports/models.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from datetime import datetime, timedelta
 import logging
 from urllib import urlencode
+from dimagi.utils.dates import DateSpan
```
Comment (dimagimon) [inline corehq/apps/reports/models.py]: F401 'DateSpan' imported but unused

File: corehq/apps/userreports/reports/specs.py [hunk 2]
```diff
@@ -227,7 +228,7 @@ class FilterSpec(JsonObject):
     This is the spec for a report filter - a thing that should show up as a UI filter element
     in a report (like a date picker or a select list).
     """
-    type = StringProperty(required=True, choices=['date', 'numeric', 'choice_list', 'dynamic_choice_list'])
+    type = StringProperty(required=True, choices=['date', 'sliding_date', 'numeric', 'choice_list', 'dynamic_choice_list'])
```
Comment (dimagimon) [inline corehq/apps/userreports/reports/specs.py]: E501 line too long (123 > 115 characters)

**Comment by czue** [comment]
@nickpell I think I don't totally understand how this is wired up. is it right that from a normal report UI perspective a `sliding_date` filter is exactly the same as a `date`? but they are somehow "special" when you view saved/scheduled reports? If so, that might be an argument for making them the same underlying thing but then putting the logic into the saved reports object (not the filter itself).

not sure if that makes complete sense though since i'm not totally sure how this all works. 

alternatively, maybe you could sprinkle in some comments?

**Comment by nickpell** [comment]
@czue Good point, this can be implemented without introducing a new filter type, by moving all the functionality to saved reports.  Will make some edits and add some comments.

**Comment by nickpell** [comment]
closing for https://github.com/dimagi/commcare-hq/pull/6685

**Pull request closed without merging**